### ISSUE TITLE
FloatingActionButton animation is gone

### ISSUE THREAD

**Issue by TheLastProject** [kontalk/androidclient#937]
### Expected behavior
On 4.0-beta5, clicking the pencil FAB produced a rather pretty animation

### Actual behavior
On 4.0-beta6.1 there is no animation anymore, the screen instantly flashes to the end state, looking rather jarring

### Steps to reproduce
1. Open Kontalk
2. Press FAB

### Environment

Kontalk version: (please also specify origin, e.g. F-Droid or Google Play)
4.0-beta6.1 (F-Droid)

Android version: (please also specify custom ROM name if applicable)
7.1.1 (CyanogenMod 14.1)

Device model:
Motorola Moto E 2015 LTE (surnia)

### Logs

If you have a system log related to the issue please send it via email at
devteam (at) kontalk.org. Please do not attach it or publish it anywhere because
it may contain sensitive information. If you have root access, you can use Catlog
(available on both Google Play and F-Droid) to produce it. Alternatively, if you
have the Android SDK installed on your computer, connect your device and execute:

```
adb logcat -d | grep `adb shell ps | grep org.kontalk | cut -c10-15` >kontalk-log.txt
```

and send us the text file.

### Other

If there is a relevant forum discussion, please link to it here.

**Comment by daniele-athome** [comment]
I've recently modified the FloatingActionButton library to respect animation scaling system settings. What are the settings on your device right now? (Android Settings > Developer Options > Drawing > Animator/Window animation scale)

**Comment by TheLastProject** [comment]
It's all the default, 1x.

![screenshot_20170124-165825](https://cloud.githubusercontent.com/assets/1885159/22255354/6dd726cc-e257-11e6-8a30-3b4bddcbd4c8.png)

**Comment by TheLastProject** [comment]
Looking at the changes you made, I'm pretty sure the issue is in Utils.java: https://github.com/kontalk/androidclient/issues/937#issuecomment-274844626

Specifically, with the getFloat calls, you default to 0. Seeing how you're working with deprecated functions and those variables may disappear or get renamed (maybe they did in my release?) it definitely makes more sense to default to 1 as it's the default for everyone who doesn't change developer options which, lets be honest, basically nobody does.

**Comment by OffifialMITX** [comment]
Changing animation speed to 10x makes this even more strange - pressing the fab causes:

Pencil rotation changes
"New chat" appears
"New group chat" appears

Pressing it again makes the same thing backwards **but** both actions are still flashing.

Am 24. Januar 2017 22:53:41 MEZ, schrieb Sylvia van Os <notifications@github.com>:
>Looking at the changes you made, I'm pretty sure the issue is in
>Utils.java:
>https://github.com/kontalk/androidclient/issues/937#issuecomment-274844626
>
>Specifically, with the getFloat calls, you default to 0. Seeing how
>you're working with deprecated functions and those variables may
>disappear or get renamed (maybe they did in my release?) it definitely
>makes more sense to default to 1 as it's the default for everyone who
>doesn't change developer options which, lets be honest, basically
>nobody does.
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/kontalk/androidclient/issues/937#issuecomment-274951017

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9/p≡p gesendet.

**Comment by daniele-athome** [comment]
> Specifically, with the getFloat calls, you default to 0

I interpreted that the default value would be returned by the system and the default of 0 was in case of errors. But [reading more carefully in the docs](https://developer.android.com/reference/android/provider/Settings.Global.html#ANIMATOR_DURATION_SCALE):

> The default value is 1.

Thanks people.

**Comment by daniele-athome** [comment]
Fixed in 60033ff93bf6df08c28d6c596388e0efe3d7cfc6.

**Comment by OffifialMITX** [comment]
Just updated to F-Droid/4.0.0 nothing changed.

**Comment by TheLastProject** [comment]
Just want to state that for my device it is fixed in 4.0.0, so it seems more specific to @OfficialMITX's setup

**Comment by OffifialMITX** [comment]
Thinking of that, I remembered that my animations are all set to 0 so i changed them and yes, it's not this bug.
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9/p≡p gesendet.
